File: leetcode/easy/1.two-sum/solutions/java/Main.java
import java.util.Arrays;

public class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();
        
        // Test Case 1
        System.out.println("Test 1:");
        int[] nums1 = {2, 7, 11, 15};
        int target1 = 9;
        int[] result1 = solution.twoSum(nums1, target1);
        System.out.println("Input: nums = " + Arrays.toString(nums1) + ", target = " + target1);
        System.out.println("Output: " + Arrays.toString(result1));
        System.out.println("Expected: [0, 1]");
        System.out.println();
        
        // Test Case 2
        System.out.println("Test 2:");
        int[] nums2 = {3, 2, 4};
        int target2 = 6;
        int[] result2 = solution.twoSum(nums2, target2);
        System.out.println("Input: nums = " + Arrays.toString(nums2) + ", target = " + target2);
        System.out.println("Output: " + Arrays.toString(result2));
        System.out.println("Expected: [1, 2]");
        System.out.println();
        
        // Test Case 3
        System.out.println("Test 3:");
        int[] nums3 = {3, 3};
        int target3 = 6;
        int[] result3 = solution.twoSum(nums3, target3);
        System.out.println("Input: nums = " + Arrays.toString(nums3) + ", target = " + target3);
        System.out.println("Output: " + Arrays.toString(result3));
        System.out.println("Expected: [0, 1]");
        System.out.println();
        
        // Test Case 4
        System.out.println("Test 4:");
        int[] nums4 = {4, 5, 6};
        int target4 = 10;
        int[] result4 = solution.twoSum(nums4, target4);
        System.out.println("Input: nums = " + Arrays.toString(nums4) + ", target = " + target4);
        System.out.println("Output: " + Arrays.toString(result4));
        System.out.println("Expected: [0, 2]");
        System.out.println();
    }
}

File: leetcode/easy/1.two-sum/solutions/go/main.go
package main

import "fmt"

func main() {
	// Test Case 1
	fmt.Println("Test 1:")
	nums1 := []int{2, 7, 11, 15}
	target1 := 9
	result1 := twoSum(nums1, target1)
	fmt.Printf("Input: nums = %v, target = %d\n", nums1, target1)
	fmt.Printf("Output: %v\n", result1)
	fmt.Println("Expected: [0, 1]")
	fmt.Println()

	// Test Case 2
	fmt.Println("Test 2:")
	nums2 := []int{3, 2, 4}
	target2 := 6
	result2 := twoSum(nums2, target2)
	fmt.Printf("Input: nums = %v, target = %d\n", nums2, target2)
	fmt.Printf("Output: %v\n", result2)
	fmt.Println("Expected: [1, 2]")
	fmt.Println()

	// Test Case 3
	fmt.Println("Test 3:")
	nums3 := []int{3, 3}
	target3 := 6
	result3 := twoSum(nums3, target3)
	fmt.Printf("Input: nums = %v, target = %d\n", nums3, target3)
	fmt.Printf("Output: %v\n", result3)
	fmt.Println("Expected: [0, 1]")
	fmt.Println()

	// Test Case 4
	fmt.Println("Test 4:")
	nums4 := []int{4, 5, 6}
	target4 := 10
	result4 := twoSum(nums4, target4)
	fmt.Printf("Input: nums = %v, target = %d\n", nums4, target4)
	fmt.Printf("Output: %v\n", result4)
	fmt.Println("Expected: [0, 2]")
	fmt.Println()
}
